### PR TITLE
FC-0001: Remove old edx rest api client

### PR DESCRIPTION
**Pull request by dyudyunov**
closes https://github.com/openedx/public-engineering/issues/42

Requires:
- https://github.com/openedx/edx-platform/pull/30240 (should be merged before this PR)
- https://github.com/openedx/devstack/pull/919

Description:
- extracted 3 base API clients:
  - `BackendServiseAPIClient` - uses the backend service user (enterprise_worker) to make calls to APIs
  - `UserAPIClient` - uses the exact user to call APIs (for API endpoints that expect the request.user to perform)
  - `NoAuthAPIClient` - for work with endpoints without authentication required
- all other clients inherit from those three (except the EcommerceApiClient which depends on edx-platform utility)
- updated the application settings for work with dot application
- updated tests

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dyudyunov! I've created [OSPR-6615](https://openedx.atlassian.net/browse/OSPR-6615) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by dyudyunov** [comment]
I've just finished running diff_cov locally to double-check what I've missed, but I have 100% diff coverage 💁
I'll add tests to pass the pipeline if somebody helps me to find what I've missed :)
```
diff-cover coverage.xml --compare-branch master
-------------
Diff Coverage
Diff: master...HEAD, staged and unstaged changes
-------------
enterprise/api_client/client.py (100%)
enterprise/api_client/discovery.py (100%)
enterprise/api_client/ecommerce.py (100%)
enterprise/api_client/enterprise.py (100%)
enterprise/api_client/enterprise_catalog.py (100%)
enterprise/api_client/lms.py (100%)
enterprise/heartbeat/checks.py (100%)
-------------
Total:   271 lines
Missing: 0 lines
Coverage: 100%
-------------
````

P.S. Adding the fix for code quality in few minutes

**Comment by dyudyunov** [comment]
there will be additional commit with tests for recent changes soon

File: enterprise/api_client/client.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        """
+        Helper method to construct the API URL.
+
+        Args:
+            path (str): the API endpoint path string.
+        """
+        return urljoin(f"{self.API_BASE_URL}/", f"{path.strip('/')}{'/' if self.APPEND_SLASH else ''}")
+
+
+class BackendServiseAPIClient(APIClientMixin):
```
Comment (dianakhuang) [inline enterprise/api_client/client.py:44]: In this name, Service is misspelled as `Servise`. All subclasses should be updated with the correct spelling as well.

**Review by dianakhuang — COMMENTED** [review]
Mostly it looks good, but I did catch one misspelling.

File: enterprise/api_client/client.py (hunk 1 above)
Comment (dyudyunov) [reply enterprise/api_client/client.py]: Fixed, thanks

**Comment by johnnagro** [comment]
@dyudyunov could you please fix the conflicts with the master branch? then we can coordinate a merge. thank you!

**Comment by dyudyunov** [comment]
> @dyudyunov could you please fix the conflicts with the master branch? then we can coordinate a merge. thank you!

hi, @johnnagro 
I rebased the branch and updated the app version, please check

**Review by johnnagro — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@dyudyunov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-enterprise&entry.867055334=https://github.com/openedx/edx-enterprise/pull/1532&entry.1484655318=https://github.com/dyudyunov&entry.752974735=2022-04-13+15:30&entry.1917517419=2022-04-21+19:30&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by natabene** [comment]
@dyudyunov @dianakhuang Thanks!

**Pull request merged**